src/server/string_family.cc

```diff
@@ -1208,18 +1195,19 @@ void StringFamily::MSet(CmdArgList args, ConnectionContext* cntx) {
     LOG(INFO) << "MSET/" << transaction->GetUniqueShardCnt() << str;
   }
 
-  atomic_bool success = true;
+  AggregateStatus result;
   auto cb = [&](Transaction* t, EngineShard* shard) {
     ShardArgs args = t->GetShardArgs(shard->shard_id());
-    OpMSet(t->GetOpArgs(shard), args, &success);
+    if (auto status = OpMSet(t->GetOpArgs(shard), args); status != OpStatus::OK)
+      result = status;
     return OpStatus::OK;
   };
 
-  OpStatus status = transaction->ScheduleSingleHop(std::move(cb));
-  if (success.load()) {
+  transaction->ScheduleSingleHop(std::move(cb));
+  if (*result == OpStatus::OK) {
     cntx->SendOk();
   } else {
-    cntx->SendError(status);
+    cntx->SendError(*result);  // docs: MSET can't fail?
   }
 }
 
```

Comment:
> cntx->SendError(status);

It's always OK, because its the status from the tx. Either this is a bug or the author knew that MSET can't fail, so decided to keep code that always sends OK 😆

Reply:
The question is what should we do, always reply ok or send an actual error?

Reply:
Oh a wow, I forgot transaction catched out-of-memory errors and returns them 😵